designs/discontinue_usage_of_kube_rbac_proxy.md

```diff
@@ -0,0 +1,285 @@
+calls for a change.
+
+In this document is proposed replacing the [kube-rbac-proxy][kube-rbac-proxy] within
+[Network Policies][k8s-doc-networkpolicies] follow-up for potentially enhancements
+to protect the metrics endpoint combined with [cert-manager][cert-manager] and a new
+feature introduced in controller-runtime, see [here][cr-pr].
+
+**For the future (when kube-rbac-proxy be part of the k8s umbrella)**, it is proposed the usage of the
+[Plugins API provided by Kubebuilder](./../docs/book/src/plugins/plugins.md),
+to create an [external plugin](./../docs/book/src/plugins/creating-plugins.md)
+to properly integrate the solution with Kubebuilder and provide a helper to allow users to opt-in as please them.
+
+## Open Questions
+
+1. [Network Policies][k8s-doc-networkpolicies] is implemented by the cluster’s CNI. Are we confident that the proposed policy is supported by all the major CNIs in use?
+
+> Besides [Network Policies][k8s-doc-networkpolicies] being part of the core Kubernetes API, their enforcement relies on the CNI plugin installed in
+the Kubernetes cluster. While support and implementation details can vary among CNIs, it seems that the most commonly used ones,
+such as Calico, Cilium, WeaveNet, and Canal, shows to offer  support for NetworkPolicies.
+>
+>Also, there was concern in the past because AWS did not support it. However, this changed,
+>as detailed in their announcement: [Amazon VPC CNI now supports Kubernetes Network Policies](https://aws.amazon.com/blogs/containers/amazon-vpc-cni-now-supports-kubernetes-network-policies/).
+>
+>Moreover, under this proposal users still able to disable/enable this option as please them.
+
+1. NetworkPolicy is a simple firewall and does not provide authn/authz and encryption.
+
+> Yes, that's correct. NetworkPolicy acts as a basic firewall for pods within a Kubernetes cluster, controlling traffic
+> flow at the IP address or port level. However, it doesn't handle authentication (authn), authorization (authz),
+> or encryption directly like kube-rbac-proxy solution.
+>
+> However, if we be able to combine the cert-manager and the new feature provided
+> by controller-runtime we can achieve the same or a superior level of protection
+> without relay in any extra third-party dependency.
+
+1. Could not Kubebuilder maintainers use the shared infrastructure to continue building and promoting those images under the new `registry.k8s.io`?
+
+> We tried to do that, see [here](https://github.com/kubernetes/test-infra/blob/master/config/jobs/image-pushing/k8s-staging-kubebuilder.yaml) the recipe implemented.
+> However, it does not work because kube-rbac-proxy is not under the
+> kubernetes umbrella. More over we experiment the GitHub Repository as alternative approach, see the [PR](https://github.com/kubernetes-sigs/kubebuilder/pull/3854) but seems
+> that we are not allowed to use it. Nevertheless, both approaches does not show to sort out all motivations and requirements
```

Comment:
> ... both approaches does not show to sort out all ...
I think there's a typo here, but I'm not sure what this is trying to say.

Reply:
It is a section of questions and answers
We are trying to clarify here that:
- We cannot use the e2e shared infra and why
- We cannot use the github packages/repository as well

Anyway, if we would be able to address the needs within any of those approaches that does NOT sort out ALL our problems and motivations. Either does not seems the best approach for the long term.

However, please feel free to let us know what is not clear and provide any suggestion.

Reply:
Got it. How about

```suggestion
> that we are not allowed to use it. Nevertheless, neither approach sorts out all motivations and requirements
```